sw/device/lib/base/global_mock.h

```diff
@@ -76,10 +76,10 @@ class GlobalMock {
   GlobalMock &operator=(GlobalMock &&) = delete;
 
  private:
-  static Mock *instance_;
+  static GlobalMock<Mock> *instance_;
```

Comment:
This also works, not sure what the best practice is?

```suggestion
  static GlobalMock *instance_;
```

Reply:
I slightly prefer the explicit type parameters here for symmetry with line 82. Omitting the type parameters on line 82 causes the following error:

```
./sw/device/lib/base/global_mock.h:82:1: error: invalid use of template-name 'global_mock::GlobalMock' without an argument list
   82 | GlobalMock *GlobalMock<Mock>::instance_ = nullptr;
      | ^~~~~~~~~~
```

I can't find anything in the OT or Google C++ style guides.  Assuming you don't feel strongly, I'll leave this as is.